Fails to read negative integer values from VHDL testbenches

When reading integer values from a VHDL testbench, negative values are treated as if they are unsigned. Note that reading an integer value does not read a BinaryValue but instead a ConstantObject. I've attached a simple example to reproduce the issue.

[integer_read_fail.zip](https://github.com/cocotb/cocotb/files/5384567/integer_read_fail.zip)

@ktbarrett regarding compilation warning for this:
> #### -Wconversion
> ... Warnings about conversions between signed and unsigned integers are disabled by default in C++ unless `-Wsign-conversion` is explicitly enabled.

When I add `-Wsign-conversion` I get:

```
cocotb\share\lib\vhpi\VhpiCbHdl.cpp: In member function 'virtual long int VhpiSignalObjHdl::get_signal_value_long()':
    cocotb\share\lib\vhpi\VhpiCbHdl.cpp:792:24: warning: conversion to 'long int' from 'vhpiIntT {aka unsigned int}' may change the sign of the result [-Wsign-conversion]
         return value.value.intg;
                            ^
```

--- a/cocotb/share/lib/vhpi/VhpiCbHdl.cpp
+++ b/cocotb/share/lib/vhpi/VhpiCbHdl.cpp
@@ -789,7 +789,7 @@ long VhpiSignalObjHdl::get_signal_value_long()
         LOG_ERROR("VHPI: Failed to get value of type long");
     }
 
-    return value.value.intg;
+    return static_cast<int32_t>(value.value.intg);
 }
 
 

--- a/documentation/source/newsfragments/2129.bugfix.rst
+++ b/documentation/source/newsfragments/2129.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed issue on Mac OS and Linux where negative integers were returned as large positive values.

--- a/documentation/source/newsfragments/2129.bugfix.rst
+++ b/documentation/source/newsfragments/2129.bugfix.rst
@@ -1,1 +1,1 @@
-Fixed issue on Mac OS and Linux where negative integers were returned as large positive values.
+Fixed an issue with VHPI on Mac OS and Linux where negative integers were returned as large positive values.

